Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs

```diff
@@ -0,0 +1,112 @@
+// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using Taarafo.Core.Models.PostImpressions;
+using Taarafo.Core.Models.PostImpressions.Exceptions;
+using Taarafo.Core.Models.Posts;
+using Taarafo.Core.Models.Profiles;
+
+namespace Taarafo.Core.Services.Foundations.PostImpressions
+{
+    public partial class PostImpressionService
+    {
+        private void ValidatePostImpressionOnAdd(PostImpression postImpression)
+        {
+            ValidatePostImpressionIsNotNull(postImpression);
+
+            Validate(
+                (Rule: IsInvalid(postImpression.PostId), Parameter: nameof(PostImpression.PostId)),
+                (Rule: IsInvalid(postImpression.Post), Parameter: nameof(PostImpression.Post)),
+                (Rule: IsInvalid(postImpression.ProfileId), Parameter: nameof(PostImpression.ProfileId)),
+                (Rule: IsInvalid(postImpression.Profile), Parameter: nameof(PostImpression.Profile)),
+                (Rule: IsInvalid(postImpression.CreatedDate), Parameter: nameof(PostImpression.CreatedDate)),
+                (Rule: IsInvalid(postImpression.UpdatedDate), Parameter: nameof(PostImpression.UpdatedDate)),
+
+                (Rule: IsNotSame(
+                    firstDate: postImpression.UpdatedDate,
+                    secondDate: postImpression.CreatedDate,
+                    secondDateName: nameof(PostImpression.CreatedDate)),
+                Parameter: nameof(PostImpression.UpdatedDate)),
+
+                (Rule: IsNotRecent(postImpression.CreatedDate), Parameter: nameof(PostImpression.CreatedDate)));
+        }
+
+        private static void ValidatePostImpressionIsNotNull(PostImpression postImpression)
+        {
+            if (postImpression is null)
+            {
+                throw new NullPostImpressionException();
+            }
+        }
+
+        private static dynamic IsInvalid(Guid id) => new
+        {
+            Condition = id == Guid.Empty,
+            Message = "Id is required."
+        };
+
+        private static dynamic IsInvalid(DateTimeOffset date) => new
+        {
+            Condition = date == default,
+            Message = "Date is required."
+        };
+
+        private static dynamic IsInvalid(Post post) => new
+        {
+            Condition = post == null,
+            Message = "Post is required."
+        };
```

Comment:
```
private static dynamic IsInvalid(object @object) => new
{
    Condition = @object  is null,
    Message = "Object is required."
};
```

Reply:
I was able to change this, but one of the validation test is now failing. The test starting in line 164 in the PostImpressionServiceTests.Validations.Add.cs file is stating that no exception is being thrown, but it is expecting the PostImpressionValidationException.

Reply:
It is working now, please ignore my last message.  I made sure to remove the periods on all of the validation messages. The updated section is in the commit named CODERUB: Profile And Post Replaced with Object. The PR link: https://github.com/hassanhabib/Taarafo.Core/pull/221